service/apigateway/internal/awsrestjson/marshal.go

```diff
@@ -19,40 +19,44 @@ type ProtoGetAPIKeyMarshaler struct {
 // This method calls appropriate marshal shape functions as per the input shape and protocol used by the service.
 func (m ProtoGetAPIKeyMarshaler) MarshalOperation(r *aws.Request) {
 	var err error
-	encoder := v2.NewEncoder(r.HTTPRequest)
+	encoder := restV2.NewEncoder(r.HTTPRequest)
 	// adds content-type header
 	encoder.AddHeader("Content-Type").String("application/json")
 
 	err = MarshalGetAPIKeyInputShapeAWSREST(m.Input, encoder)
 	if err != nil {
 		r.Error = err
+		return
 	}
+	encoder.Encode()
```

Comment:
If `Encode` returns an `error` need to check and return it if so.

Reply:
Sure, `Encode` doesn't directly return an error, but returns a pointer to http.request. I will check if r.Error != nil and return if not nil